src/backend/distributed/commands/schema.c

```diff
@@ -53,6 +84,32 @@ PreprocessDropSchemaStmt(Node *node, const char *queryString,
 						 ProcessUtilityContext processUtilityContext)
 {
 	DropStmt *dropStatement = castNode(DropStmt, node);
+	Assert(dropStatement->removeType == OBJECT_SCHEMA);
+
+	if (!ShouldPropagate())
+	{
+		return NIL;
+	}
+
+	EnsureCoordinator();
+
+	bool includesDistributedServer =
+		NameListHasDistributedSchema(dropStatement->objects);
+
+	if (!includesDistributedServer)
+	{
+		return NIL;
+	}
+
+	if (list_length(dropStatement->objects) > 1)
+	{
+		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+						errmsg("cannot drop distributed schema with other schemas"),
+						errhint("Try dropping each object in a separate DROP command")));
```

Comment:
could you explain here, and possibly in a comment on the function why we disallow the dropping of multiple schema's in 1 command?

If you look at the implementation of other drop commands you will see that dropping multiple objects at once is supported.

We achieve this by filtering the list of schema's to drop by the once that are distributed. Before we deparse we use this filtered list, and restore the original list thereafter to locally drop all listed schema's. Any reason that doesn't work here?

Reply:
This is a big outstanding question before we can merge the PR.